### PR TITLE
Added memory info to horizontal autoscaler docs

### PR DESCRIPTION
**Pull request by bfallonf**
Redo of #4498 

cc: @DirectXMan12 

Ready for peer review. @openshift/team-documentation PTAL.

### CODE REVIEW

File: dev_guide/pod_autoscaling.adoc [hunk 1]
```diff
@@ -118,6 +119,64 @@ spec:
 <7> The percentage of the requested CPU that each pod should ideally be using
 ====
 
+[[pod-autoscaling-memory]]
+== Autoscaling for Memory Utilization
+
+Use the `oc autoscale` command and specify the maximum number of pods to run at
```
Comment (DirectXMan12) [inline dev_guide/pod_autoscaling.adoc]: might want to remove this sentence, considering the sentence below ("Unlike CPU-based autoscaling").

**Review by DirectXMan12 — COMMENTED** [review]
one small comment, otherwise LGTM

File: dev_guide/pod_autoscaling.adoc [hunk 2]
```diff
@@ -118,6 +119,64 @@ spec:
...
+any given time. Optionally, you can specify the minimum number of pods and the
+average memory utilization your pods should target as well, otherwise those are
+given default values from the {product-title} server.
+
+. Memory-base autoscaling is only available with the `v2alpha1` version of the
```
Comment (mburke5678) [inline dev_guide/pod_autoscaling.adoc]: s/Memory-base/Memory-based

File: dev_guide/pod_autoscaling.adoc (hunk 1 above)
Comment (mburke5678) [inline dev_guide/pod_autoscaling.adoc]: +1

**Comment by mburke5678** [comment]
@bfallonf Couple of comments.

**Comment by bfallonf** [comment]
Thanks @DirectXMan12 @mburke5678 . I've changed as suggested.

File: dev_guide/pod_autoscaling.adoc [hunk 3]
```diff
@@ -71,7 +74,7 @@ directly to the replica count of the deployment configuration. Note that autosca
 applies only to the latest deployment in the `Complete` phase.
 
 [[creating-a-hpa]]
-== Creating a Horizontal Pod Autoscaler
+== Autoscaling for CPU Utilization
```
Comment (chengzhang1016) [inline dev_guide/pod_autoscaling.adoc]: HPA V2 support both cpu and memory resource metrics, you need to describe customer can use both HPA v1 and v2 to create a hpa Autoscaling for cpu utilization in here.

File: dev_guide/pod_autoscaling.adoc [hunk 4]
```diff
@@ -45,8 +45,11 @@ The following metrics are supported by horizontal pod autoscalers:
 
 |Metric |Description
```
Comment (chengzhang1016) [inline dev_guide/pod_autoscaling.adoc]: A useful doc for your reference: https://github.com/DirectXMan12/kubernetes.github.io/blob/7023dc32c507f0edf8ab7278ebedd9a177a00396/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough.md

**Comment by bfallonf** [comment]
Thanks for the comments @chengzhang1016 . I've added another section about CPU and v2. Please let me know if there's anything else for this PR.

cc @DirectXMan12

**Comment by bfallonf** [comment]
[rev_history]
|xref:../dev_guide/pod_autoscaling.adoc#dev-guide-pod-autoscaling[Pod Autoscaling]
|Added the xref:../dev_guide/pod_autoscaling.adoc#pod-autoscaling-memory[Autoscaling for Memory Utilization] and changed parts of the section to be specific to CPU Utilization. 
%

**Comment by gaurav-nelson** [comment]
@DirectXMan12 @chengzhang1016 is `Memory-based autoscaling` is a technical preview feature? PS:https://bugzilla.redhat.com/show_bug.cgi?id=1501483

**Comment by DirectXMan12** [comment]
It's still tech preview at least up to v3.9, since it's an alpha API.  The API has moved to beta 3.9.  However, memory-based autoscaling is *really* tricky.  We should have a note about that in the docs (you need to make sure that your application's memory usage can ever actually go down, which is more complicated than it seems).

**Pull request merged**